api/proto/metrics/metrics.proto

```diff
@@ -24,6 +24,42 @@ message memoryMetric {
   double availPercentage = 5;
 }
 
+// App Memory Metric
+message AppMemoryMetric {
```

Comment:
Why do we need new messages with the associated compatibility and transition?
Can't we just explain what the existing counters mean for the applications vs. system?

In general we have total = used + available so the fact that we currently only report used and available is no loss of information. Might have been better to report total and used, but doesn't make sense to change the API for that; just clarify with comments and make sure the code is correct.

Reply:
I split up AppMemoryMetric from DeviceMemoryMetric because these can grow in their own ways. 

For example, For App Instances, we can add Eve overhead for that app instance. K3s/Azure modules might provide more info that we may want to add. 

For Device Memory, we can provide more insights into usage of various components.

Hence proposed not using a common structure. IMHO, this makes it clean and more easy to enhance.

Reply:
But that causes an immediate cost to have the controller look in both places to handle the transition. If the fields we need are the same then it doesn't seem worth it.

Reply:
You are actually deprecating all of memoryMetric, so perhaps the comment should be for the message?

Reply:
Done. Added the comment to Memory metric as well./